benchmark/src/benchmark.ts

```diff
@@ -0,0 +1,82 @@
+import { isEqual } from 'lodash-es';
+
+export interface BenchmarkValue<T> {
+    /**
+     * The value returned by the benchmarked function
+     */
+    value: T;
+    /**
+     * The average time it took to run the function in ms
+     */
+    time: number;
+}
+
+/**
+ * @param functionToBenchmark the deterministic function that should be benchmarked, it will be run multiple times
+ * @param options additional options for the benchmark, the defaults are {@link defaultOptions}
```

Comment:
Maybe this comment could be a bit better. My first thought reading it was that this means code like `options = options ?? defaultOptions`, but it's rather a merge of the options.

Reply:
If I would have meant `options = options ?? defaultOptions` I would have written `the default is {@link defaultOptions}`.
But I improved the documentation to be more clear.

Reply:
Oh, that's not a big difference between `defaults are` and `default is`. But it's better now.